nomos-services/system-sig/src/lib.rs

```diff
@@ -0,0 +1,74 @@
+// std
+
+// crates
+use futures::stream::StreamExt;
+use log::error;
+// internal
+use overwatch_rs::overwatch::handle::OverwatchHandle;
+use overwatch_rs::services::handle::ServiceStateHandle;
+use overwatch_rs::services::life_cycle::LifecycleMessage;
+use overwatch_rs::services::relay::NoMessage;
+use overwatch_rs::services::state::{NoOperator, NoState};
+use overwatch_rs::services::{ServiceCore, ServiceData, ServiceId};
+use overwatch_rs::DynError;
+
+pub struct SystemSig {
+    service_state: ServiceStateHandle<Self>,
+}
+
+impl SystemSig {
+    async fn should_stop_service(msg: LifecycleMessage) -> bool {
+        match msg {
+            LifecycleMessage::Shutdown(sender) => {
+                // TODO: Try to finish pending transactions if any and close connections properly
```

Comment:
I believe there won't be any pending txs in this service

Reply:
hopefully not 😅